### PR TITLE
EPMDJ-7763 Requirements for Admin backend

### PR DESCRIPTION
**Pull request by LikhanovMaxim**
added metadata for agents: count and size of classes

### CODE REVIEW

File: core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt [hunk 1]
```diff
@@ -94,17 +113,17 @@ class AgentEndpoints(override val kodein: Kodein) : KodeinAware {
...
-                            configHandler.updateAgent(agentId, parameters)
-                            logger.debug { "Agent with id '$agentId' was updated successfully" }
-                            HttpStatusCode.OK to EmptyContent
-                        } ?: HttpStatusCode.NotFound to ErrorResponse("agent '$agentId' not found")
+                        val newParameters = HashMap(storageParameters)
+                        parameters.forEach { updateParameter ->
+                            newParameters[updateParameter.key]?.let {
+                                newParameters[updateParameter.key] = it.copy(value = updateParameter.value)
+                            } ?: logger.warn { "cannot find and update the parameter '$updateParameter'" }
+                        }
```
Comment (Mazel-Tovr) [inline core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt]:
```suggestion
                        val newParameters = storageParameters.toMutableMap()
                        parameters.forEach { (key, value) ->
                            newParameters[key]?.let {
                                newParameters[key] = it.copy(value = value)
                            } ?: logger.warn { "Cannot find and update the parameter '$key'" }
                        }
```

File: core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt [hunk 2]
```diff
@@ -94,17 +113,17 @@ class AgentEndpoints(override val kodein: Kodein) : KodeinAware {
...
+                        configHandler.store(agentId, newParameters)
+                        configHandler.updateAgent(agentId, parameters)
```
Comment (Mazel-Tovr) [inline core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt]: why we store "newParameters" but updateAgent with old one ?

File: core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt [hunk 3]
```diff
@@ -74,6 +76,23 @@ class AgentEndpoints(override val kodein: Kodein) : KodeinAware {
...
+                    .responds(
+                        ok<String>()
+                    )
+                get<ApiRoot.Agents.Metadata>(meta) {
+                    val metadataAgents = agentManager.all().map { it.id }.map {
+                        agentManager.adminData(it).buildManager.agentBuilds.map { agentBuild ->
+                            val agentKey = AgentKey(it, agentBuild.info.version)
+                            mapOf(agentKey.toString() to stores[it].loadMetadata(agentKey))
+                        }
+                    }.flatten()
```
Comment (Mazel-Tovr) [inline core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt]:
```suggestion
                    val metadataAgents = agentManager.all().map { it.id }.flatMap {
                        agentManager.adminData(it).buildManager.agentBuilds.map { agentBuild ->
                            val agentKey = AgentKey(it, agentBuild.info.version)
                            mapOf(agentKey.toString() to stores[it].loadMetadata(agentKey))
                        }
                    }
```

Comment (Mazel-Tovr) [inline core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt]: or 
```suggestion
                    val metadataAgents = agentManager.all().flatMap {
                        agentManager.adminData(it.id).buildManager.agentBuilds.map { agentBuild ->
                            val agentKey = AgentKey(it.id, agentBuild.info.version)
                            mapOf(agentKey.toString() to stores[it.id].loadMetadata(agentKey))
                        }
                    }
```

File: core/src/main/kotlin/endpoints/AgentManager.kt [hunk 4]
```diff
@@ -609,6 +607,14 @@ suspend fun AgentWsSession.setPackagesPrefixes(prefixes: List<String>) =
 suspend fun AgentWsSession.triggerClassesSending() =
     sendToTopic<Communication.Agent.LoadClassesDataEvent, String>("").await()
 
+internal suspend fun StoreClient.storeMetadata(agentKey: AgentKey, metadata: Metadata) {
+    store(StoredMetadata(agentKey.toString(), metadata))
+}
+
+internal suspend fun StoreClient.loadMetadata(agentKey: AgentKey) =
+    findById<StoredMetadata>(agentKey.toString())?.data ?: Metadata()
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/endpoints/AgentManager.kt]: formatting

File: core/src/main/kotlin/endpoints/agent/AgentHandler.kt [hunk 5]
```diff
@@ -121,7 +122,7 @@ class AgentHandler(override val kodein: Kodein) : KodeinAware {
                             }
 
                             MessageType.FINISH_CLASSES_TRANSFER -> adminData.apply {
-                                initClasses()
+                                initClasses(agentInfo.buildVersion)
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/endpoints/agent/AgentHandler.kt:125]: agentKey

**Pull request merged**